oresat_dxwifi/resources/temperature.py

```diff
@@ -0,0 +1,114 @@
+# Purpose:
+#       Issue: https://github.com/oresat/oresat-dxwifi-software/issues/1.
+#       The temperature "read" by an NTC thermistor is calculated based on
+#           the voltage recieved at the ADC input pin.
+#       The calculations are based on:
+#           https://docs.google.com/document/d/16b_Qd6Pj2IAk5HQQl7JAP1T9Qd6QbKICMd-0swd9L6Y/edit
+#       The resource then sends the calculated temperature to the Object Directory upon
+#           resource start. An SDO read callback is also implemented
+
+from olaf import Resource, Adc, logger
+import math as m
+
+
+class TemperatureResource(Resource):
+    """Resource for getting the temperature of the NTC thermistor connected at AIN6"""
+
+    # Olaf Constants ------------------------------------------------------------------------------
+
+    TEMPERATURE_INDEX = 0x6001
+
+    # Resource Related ----------------------------------------------------------------------------
+
+    def __init__(self, adc_thermistor_pin=6, is_mock_adc=False):
+        """Sets up the ADC. Determines which pin to use and whether the ADC is a real or mock ADC.
+
+        Args:
+            adc_thermistor_pin (int, optional): Pin where the voltage is read. Defaults to 6.
+            is_mock_adc (bool, optional): False=real world ADC, True=mock ADC. Defaults to False.
+        """
+        super().__init__()
+        self.index = self.TEMPERATURE_INDEX
+        self.adc = Adc(adc_thermistor_pin, is_mock_adc)
+
+    def on_start(self):
+        """Sets up an SDO read callback"""
+        self.node.add_sdo_read_callback(self.index, self.on_read)
+
+    def on_read(self, index: int, subindex: int) -> float:
+        """Returns temperature value
+
+        Args:
+            index (int): Index in Object Dictionary
+            subindex (int): Subindex of "index" argument
+
+        Returns:
+            float: Calculated temperature value
+        """
+        ret = None
+
+        if index == self.index:
+            ret = self.find_temperature()
+
+        return ret
+
+    # Temperature Calculation Constants -----------------------------------------------------------
+
+    KELVIN_OFFSET = 273.15
+
+    # Thermistor values at specific standard temperature (25 C)
```

Comment:
I would say `s/specific standard/rated/`. I'd personally rename these variables from `*25` to `*_rated`, so that the meaning is clearer at a glance where they're used -- the Google doc uses both `T_{R}` (etc.) and `T_{25}` (etc.), so there's no standard notation here...

Reply:
My comment might have been a bit of a misnomer. The "standard" was meant to say that is what the manufacturer uses for these values. The standard is technically arbitrary as the thermistors behavior and values change based on temperature. However, you do need to know one set of those values for the calculation we need to do. The PA thermistor document and most of the supporting documentation use subscripts. When they don't, they use T25, and B25 or the full version B25/85 etc.